pkg/controllers/leaderworkerset_controller.go

```
@@ -135,7 +138,14 @@ func (r *LeaderWorkerSetReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{}, err
 	}
 
-	if err := r.SSAWithStatefulset(ctx, lws, leaderSts, partition, replicas, revisionutils.GetRevisionKey(revision)); err != nil {
+	if !lwsUpdated {
+		// in the cases where lws is updated, the Partition in leaderSts is zero, not a useful event to record
+		if leaderSts != nil && partition != *leaderSts.Spec.UpdateStrategy.RollingUpdate.Partition {
+			r.Record.Eventf(lws, corev1.EventTypeNormal, "GroupsAreUpgrading", fmt.Sprintf("Upgrading replicas %d to %d", *leaderSts.Spec.UpdateStrategy.RollingUpdate.Partition, partition))
+		}
+	}
+
+	if err := r.SSAWithStatefulset(ctx, lws, partition, replicas, revisionutils.GetRevisionKey(revision)); err != nil {
```

can we add an event here such that if the leaders sts is nil and we have an error here, the event logs the sts creation failure.

---

I guess this is addressed by the patch failure event, but I would suggest to have this here instead of logging an event for every patch failure. Also, we should ignore logging the error if it was because the sts already exists.

---

It is patching, there is no case where it already exist right? I'll add that to the worker statefulset one though

---

sg